pkg/audit/manager.go

```diff
@@ -1061,19 +1064,33 @@ func emitEvent(constraint *unstructured.Unstructured,
 		logging.ResourceNamespace:    rnamespace,
 		logging.ResourceName:         rname,
 	}
+
 	reason := "AuditViolation"
-	ref := getViolationRef(gkNamespace, resourceGroupVersionKind.Kind, rname, rnamespace, constraint.GetKind(), constraint.GetName(), constraint.GetNamespace())
+	enamespace := gkNamespace
+	if *auditEventsInvolvedNamespace && len(rnamespace) > 0 {
+		enamespace = rnamespace
+	}
+
+	ref := getViolationRef(enamespace, resourceGroupVersionKind.Kind, rname, rrv, ruid)
 
-	eventRecorder.AnnotatedEventf(ref, annotations, corev1.EventTypeWarning, reason, "Timestamp: %s, Resource Namespace: %s, Constraint: %s, Message: %s", timestamp, rnamespace, constraint.GetName(), message)
+	if *auditEventsInvolvedNamespace || len(rnamespace) == 0 {
```

Comment:
should `auditEventsInvolvedNamespace` be false here?

Reply:
same as below, it's just to avoid printing the resources namespace if the events are being generated in the same namespace (redundant as we already know) or there is no namespace (cluster scoped resources).

Reply:
For the case where the feature is not enabled, we should keep it the same as before to reduce impact for existing users. Can you restore it back to what it was?

Reply:
done, behaviour is reverted when flag disabled